dfp/decimal/decimal

```diff
@@ -180,6 +180,286 @@ public:
+	}
+      return __sav;
+    }
+  }
+
+  template<>
+    void
+    __convert_to_v(const char* __s, decimal::decimal32& __v, ios_base::iostate& __err,
+		   const __c_locale&) throw()
+    {
+      // Assumes __s formatted for "C" locale.
+      const char* __sav = __set_C_locale();
+      if (!__sav)
+	{
+	  __err = ios_base::failbit;
+	  return;
+	}
+      char* __sanity;
+      bool __overflow = false;
+
+      __v = strtod32(__s, &__sanity);
+
+      // _GLIBCXX_RESOLVE_LIB_DEFECTS
+      // 23. Num_get overflow result.
+      if (__sanity == __s || *__sanity != '\0')
+	{
+	  __v = 0.0df;
+	  __err = ios_base::failbit;
+	}
+      else if (__overflow
+	       || __v == decimal::decimal32 {__builtin_infd32()}
+	       || __v == decimal::decimal32 {-__builtin_infd32()}
+	      )
+	{
+	  if (__v > decimal::decimal32 {0.0df})
+	    __v = DEC32_MAX;
+	  else
+	    __v = -DEC32_MAX;
+	  __err = ios_base::failbit;
+	}
+
+      setlocale(LC_ALL, __sav);
+      delete [] __sav;
+    }
+
+  template<>
+    void
+    __convert_to_v(const char* __s, decimal::decimal64& __v, ios_base::iostate& __err,
+		   const __c_locale&) throw()
+    {
+      // Assumes __s formatted for "C" locale.
+      const char* __sav = __set_C_locale();
+      if (!__sav)
+	{
+	  __err = ios_base::failbit;
+	  return;
+	}
+      char* __sanity;
+
+      __v = strtod64(__s, &__sanity);
+
+      // _GLIBCXX_RESOLVE_LIB_DEFECTS
+      // 23. Num_get overflow result.
+      if (__sanity == __s || *__sanity != '\0')
+	{
+	  __v = decimal::decimal64 {0.0dd};
+	  __err = ios_base::failbit;
+	}
+      else if (
+	       __v == decimal::decimal64 {__builtin_infd64()}
+	       || __v == decimal::decimal64 {-__builtin_infd64()} )
+	{
+	  if (__v > decimal::decimal64 {0.0dd})
+	    __v = DEC64_MAX;
+	  else
+	    __v = -DEC64_MAX;
+	  __err = ios_base::failbit;
+	}
+
+      setlocale(LC_ALL, __sav);
+      delete [] __sav;
+    }
+
+  template<>
+    void
+    __convert_to_v(const char* __s, decimal::decimal128& __v,
+		   ios_base::iostate& __err, const __c_locale&) throw()
+    {
+      // Assumes __s formatted for "C" locale.
+      const char* __sav = __set_C_locale();
+      if (!__sav)
+	{
+	  __err = ios_base::failbit;
+	  return;
+	}
+
+      char* __sanity;
+      __v = strtod128(__s, &__sanity);
+
+      // _GLIBCXX_RESOLVE_LIB_DEFECTS
+      // 23. Num_get overflow result.
+      if (__sanity == __s || *__sanity != '\0')
+	{
+	  __v = decimal::decimal128 {0.0dl};
+	  __err = ios_base::failbit;
+	}
+       else if (
+	  __v == decimal::decimal128 {__builtin_infd128()}
+	  || __v == decimal::decimal128 {-__builtin_infd128()} )
+	{
+	  if (__v > decimal::decimal128 {0.0dl})
+	    __v = DEC128_MAX;
+	  else
+	    __v = -DEC128_MAX;
+	  __err = ios_base::failbit;
+	}
+
+      setlocale(LC_ALL, __sav);
+      delete [] __sav;
+    }
+
+
+
+  template<typename _CharT, typename _InIter = std::istreambuf_iterator<_CharT>>
+  class dfp_num_get : public num_get<_CharT, _InIter> {
```

Comment:
I don't understand the naming scheme in this code. You're using reserved names like `__save_c_locale()` as though you're part of the C++ implementation (which I suppose this library is intended to be, as an implementation of an ISO TS). But then you're just adding new names like `dfp_num_get` and `dfp_basic_istream` which are not in the implementer's namespace, and so would clash with user macros using those names.

Either stop using `__reserved` names for parameters, local variables etc. or use them for everything, and don't add new non-reserved names that are not part of the C++ standard or the decimal types TR. The draft I have of TR 24733 calls this `extended_num_get`.